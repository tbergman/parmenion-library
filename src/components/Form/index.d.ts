export * from './Button/Button.d';
export * from './ButtonGroup/ButtonGroup.d';
export * from './Checkbox/Checkbox.d';
export * from './CheckboxGroup/CheckboxGroup.d';
export * from './Form/Form.d';
export * from './IconButton/IconButton.d';
export * from './InputGroup/InputGroup.d';
export * from './InputPassword/InputPassword.d';
export * from './InputStatic/InputStatic.d';
export * from './InputText/InputText.d';
export * from './Label/Label.d';
export * from './Radio/Radio.d';
export * from './RadioGroup/RadioGroup.d';
export * from './Select/Select.d';
export * from './Textarea/Textarea.d';
