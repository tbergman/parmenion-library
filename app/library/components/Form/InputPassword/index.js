export default from './InputPassword.jsx';