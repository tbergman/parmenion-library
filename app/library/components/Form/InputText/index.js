export default from './InputText.jsx';