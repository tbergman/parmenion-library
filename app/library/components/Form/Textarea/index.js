export default from './Textarea.jsx';