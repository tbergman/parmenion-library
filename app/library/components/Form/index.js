export * from './Button';
export * from './ButtonGroup';
export * from './Checkbox';
export * from './Group';
export * from './IconButton';
export * from './InputGroup';
export * from './InputPassword';
export * from './InputText';
export * from './InputStatic';
export * from './Label';
export * from './Radio';
export * from './Select';
export * from './Textarea';
