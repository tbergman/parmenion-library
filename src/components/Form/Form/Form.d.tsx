export interface FormProps {
  children?: any;
  horizontal?: boolean;
}

interface FormComponent extends React.ComponentClass<FormProps> {
  Field: typeof FormField;
  Group: typeof FormGroup;
  Row: typeof FormRow;
}

export declare const Form: FormComponent;

export interface FormFieldProps {
  children?: any;
  label?: string;
}

export interface FormRowProps {
  children?: any;
  divisions?: number;
}

export declare const FormField: React.StatelessComponent<FormFieldProps>;

export declare const FormGroup: React.StatelessComponent<FormFieldProps>;

export declare const FormRow: React.StatelessComponent<FormRowProps>;
