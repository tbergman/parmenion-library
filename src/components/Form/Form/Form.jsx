import React, { PropTypes } from 'react';
import { withContext, getContext } from 'recompose';
import { Stacked, Row, Column } from '../../Layout';
import { Label } from '../Label';

const FORM_HORIZONTAL_CONTEXT_KEY = '__formIsHorizontal';
const formContextTypes = { [FORM_HORIZONTAL_CONTEXT_KEY]: PropTypes.bool };

export const Form = withContext(
  formContextTypes,
  props => ({ [FORM_HORIZONTAL_CONTEXT_KEY]: props.horizontal }),
)(({ horizontal, ...props }) => (
  <form {...props}>
    <Stacked>
      {props.children}
    </Stacked>
  </form>
));

Form.Field = getContext(formContextTypes)((props) => {
  if (props[FORM_HORIZONTAL_CONTEXT_KEY]) {
    return <HorizontalFormField {...props} />;
  }

  return (
    <Stacked spaceBetween={0.5}>
      {props.label != null && (
        <Label>{props.label}</Label>
      )}

      {props.children}
    </Stacked>
  );
});

const HorizontalFormField = (props) => {
  if (!props.label) {
    return (
      <Stacked>
        {props.children}
      </Stacked>
    );
  }

  return (
    <Row alignItems="center">
      <Column sm={2}>
        <Label>{props.label}</Label>
      </Column>
      <Column sm={10}>
        <Stacked>
          {props.children}
        </Stacked>
      </Column>
    </Row>
  );
};

Form.Group = props => (
  <Form.Field label={props.label}>
    <Form.Row {...props}>
      {props.children}
    </Form.Row>
  </Form.Field>
);

Form.Row = props => (
  <Row divisions={props.divisions != null ? props.divisions : React.Children.count(props.children)}>
    {React.Children.map(props.children, child =>
      child != null
      ? (
        <Column md={1}>
          {child}
        </Column>
      )
      : null,
    )}
  </Row>
);

Form.propTypes = {
  children: React.PropTypes.node,
};
