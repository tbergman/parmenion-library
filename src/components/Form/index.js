export * from './Button';
export * from './ButtonGroup';
export * from './Checkbox';
export * from './CheckboxGroup';
export * from './DatePicker';
export * from './Form';
export * from './IconButton';
export * from './InputGroup';
export * from './InputPassword';
export * from './InputText';
export * from './InputStatic';
export * from './Label';
export * from './RadioGroup';
export * from './Radio';
export * from './Select';
export * from './Textarea';
