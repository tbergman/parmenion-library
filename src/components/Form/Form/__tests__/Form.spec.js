/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { Form } from '../Form';

it('renders a <Form /> correctly', () => {
  const tree = shallow(
    <Form>
      Hello, world!
    </Form>,
  );

  expect(tree).toMatchSnapshot();
});

it('renders a <Row /> correctly', () => {
  const tree = shallow(
    <Form.Row>
      <Form.Field label="Field 1">
        <div />
      </Form.Field>
      <Form.Field label="Field 2">
        <div />
      </Form.Field>
      <Form.Field label="Field 3">
        <div />
      </Form.Field>
    </Form.Row>,
  );

  expect(tree).toMatchSnapshot();
});
