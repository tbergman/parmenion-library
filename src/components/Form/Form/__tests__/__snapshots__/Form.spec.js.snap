exports[`test renders a <Form /> correctly 1`] = `
<form>
  <Stacked
    spaceBetween={1}>
    Hello, world!
  </Stacked>
</form>
`;

exports[`test renders a <Row /> correctly 1`] = `
<Row
  divisions={3}>
  <Column
    md={1}>
    <getContext(Component)
      label="Field 1">
      <div />
    </getContext(Component)>
  </Column>
  <Column
    md={1}>
    <getContext(Component)
      label="Field 2">
      <div />
    </getContext(Component)>
  </Column>
  <Column
    md={1}>
    <getContext(Component)
      label="Field 3">
      <div />
    </getContext(Component)>
  </Column>
</Row>
`;
